pkg/webhook/resources/virtualmachineimage/validator.go

```diff
@@ -70,8 +70,12 @@ func (v *virtualMachineImageValidator) Create(request *types.Request, newObj run
 
 	if newImage.Spec.SourceType == v1beta1.VirtualMachineImageSourceTypeDownload && newImage.Spec.URL == "" {
 		return werror.NewInvalidError(`url is required when image source type is "download"`, "spec.url")
-	} else if newImage.Spec.SourceType == v1beta1.VirtualMachineImageSourceTypeUpload && newImage.Spec.URL != "" {
-		return werror.NewInvalidError(`url should be empty when image source type is "upload"`, "spec.url")
+	} else if newImage.Spec.SourceType != v1beta1.VirtualMachineImageSourceTypeDownload && newImage.Spec.URL != "" {
+		return werror.NewInvalidError(`url should be empty when image source type is not "download"`, "spec.url")
+	} else if newImage.Spec.SourceType == v1beta1.VirtualMachineImageSourceTypeExportVolume {
+		if _, err := v.pvcCache.Get(newImage.Namespace, newImage.Spec.VolumeName); err != nil {
+			return werror.NewInvalidError(`volumename should be valid when image source type is "export-from-volume"`, "spec.volumename")
```

Comment:
Print the original err in the message. It could fail for other reasons even if the name is valid

Reply:
@gitlawr updated